presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1498,7 +1491,39 @@ else if (expression instanceof LongLiteral) {
         return analyzedGroupingSets;
     }
 
-    private List<FieldOrExpression> analyzeGroupingColumns(List<Expression> groupingColumns, QuerySpecification node, RelationType tupleDescriptor, AnalysisContext context, List<FieldOrExpression> outputExpressions)
+    private List<Set<Expression>> computeGroupingSetsCrossProduct(List<List<Set<Expression>>> enumeratedGroupingSets, boolean isDistinct)
```

Comment:
If there are multiple grouping sets, their cross product is used. For example, if my grouping sets were `GROUPING SETS((a,b), (c)) and GROUPING SETS ((d), (e, f))`, the resultant grouping set is `GROUPING SETS ((a,b,d), (a,b,e,f), (c,d), (c,e,f))`